backend/dataall/modules/datasets/cdk/env_role_dataset_s3_policy.py

```diff
@@ -70,18 +70,16 @@ def _generate_dataset_statements(datasets: List[Dataset]):
 
     @staticmethod
     def _set_allowed_kms_keys_statements(datasets):
-        allowed_buckets_kms_keys = []
+        imported_kms_alias = []
+        # Datasets belonging to a team and an environment are present in same region and aws account
+        imported_dataset_resources = [f"arn:aws:kms:{datasets[0].region}:{datasets[0].AwsAccountId}:key/*"]
```

Comment:
I think this can cause errors if there are no datasets. Can we move line 75 after we check if there are datasets in line 76?

Reply:
From the code flow no need according to me . Because this function is called at line 66. Which is only called when there are some datasets which are present. 
But terms of making this function more clean and error resistant if its called somewhere else I will change this as per you recommendation